wg-api/spec-documents/clipboard-getnativeformat.md

```diff
@@ -0,0 +1,35 @@
+# `clipboard.getNativeFormat(portableFormat)`
+
+## Summary
+A new `clipboard.getNativeFormat(portableFormat)` API method is introduced. It returns a `String` representing the underlying _native_ clipboard type for the supplied _portable_ format.
+
+## Platforms
+macOS, Windows, Linux
+
+## Impetus
+
+Each operating system supported by Electron utilizes a different convention to identify the data format of the clipboard contents: macOS uses [UTIs](https://en.wikipedia.org/wiki/Uniform_Type_Identifier), Linux uses [MIME types](https://en.wikipedia.org/wiki/MIME) and Windows uses [numeric values](https://docs.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats).
+
+To increase portability, Electron's `clipboard` API exposes read/write access to several high-level _portable_ clipboard formats (via `write()`, `writeText()`, `readImage()` etc). Lower-level access to the _native_ formats is also provided via `Buffer`-based read/write APIs (via `readBuffer()`, `writeBuffer()`)
+
+There's however, no currently available mechanism to query the underlying low-level _native_ format for a given high-level _portable_ format. Applications that want to have `Buffer` access to these formats are therefore required to make use of hardcoded, platform-specific string contants.
+
+The introduction of the [proposed](clipboard-writebuffers.md) `writeBuffers()` API (that allows simultaneous writes to multiple low-level clipboard formats) makes this issue more prominent, since  Chromium's underlying `ScopedClipboardWriter` implementation currently disallows simultaneously writing `portable` and `native` formats on the same call.
+
+## API Design
+
+The new API is comprised of a single method:
+
+```markdown
+### `clipboard.getNativeFormat(portableFormat)`
+
+* `portableFormat` String - One of the supported native formats: `text`, `bookmark`, `html`, `rtf`, `image`
+
+Returns `String` - An opaque, platform-dependent value representing a clipboard format (e.g. `text/plain`, `public.text` or `13`)
```

Comment:
Presumably the only useful application of this value would be to pass it into the new `clipboard.writeBuffers` (or existing `clipboard.writeBuffer`) function, yes?

So this would be used like:
```js
clipboard.writeBuffers({
  [clipboard.getNativeFormat('text')]: Buffer.from(...),
  [myCustomFormat[os.platform()]]: Buffer.from(...),
})
```

That seems a little awkward to me, though I'm not sure I have a better suggestion. Here are some random thoughts, maybe useful, maybe not.

1. Given that custom formats will also necessarily be platform-specific, how do we expect users of the API to handle those? Should we have some examples to guide best practices?
2. These values are effectively constants. Should we expose them as such? e.g. `clipboard.format.TEXT`
3. Could we use JS Symbols effectively here...? e.g. if `clipboard.format.TEXT` was a special symbol, that could distinguish it from custom formats while avoiding the collision problem.